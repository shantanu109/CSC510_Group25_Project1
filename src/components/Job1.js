

import React, { Component } from 'react';
import { connect } from 'react-redux';



class Job1 extends Component {
    
    

      
    
    render() {
        const { job } = this.props;
        const {menu } = this.props;
        const {user} = this.props.auth;

        const { isLoggedIn } = this.props.auth;
        return (
          
            <div className="post" key={menu.restname} style={{width:'50vw',marginLeft:'50px'}}>
              {1>0? 
            <div className="post-header">
              


                  <div >
                <h4 style={{display:'inline-block',marginTop:'-12px'}}>Dish : </h4> 
                <span style={{marginLeft:'10px'}}>{menu.menuname}</span>
                  </div>

                  <div >
                    
                <h4 style={{display:'inline-block',marginTop:'-12px'}}>Price : </h4> 
                <span style={{marginLeft:'10px'}}>{menu.costmenu}</span>
                  </div>

                  
                  
                  <div >
                    <h4 style={{display:'inline-block',marginTop:'-12px'}}>Ingredients</h4> 
                  <div className='ingredient_container'>
                  <div className='ingredient_tab'>
                  <div className='field'>
                            <span style={{marginLeft:'10px',fontWeight:"bold"}}>No</span>
                          </div>
                          <div className='field'>
                            <span style={{marginLeft:'10px',fontWeight:"bold"}}>Name</span>
                          </div>
                          <div className='field'>
                            <span style={{marginLeft:'10px',fontWeight:"bold"}}>Quantity</span>
                          </div>
                        </div>
                  {
                    menu && menu.ingredients.map((ingredient,index)=>{
                      return(
                        <div className='ingredient_tab'>
                          <div key={index} className='field'>
                            <span style={{marginLeft:'10px'}}>{index+1}</span>
                          </div>
                          <div key={index} className='field'>
                            <span style={{marginLeft:'10px'}}>{ingredient.inventory_id.itemname}</span>
                          </div>
                          <div key={index} className='field'>
                            <span style={{marginLeft:'10px'}}>{ingredient.quantity}</span>
                          </div>
                        </div>
                      )
                    })
                  }
                  </div>
                  </div> 
                
                
              
            
            </div> : []}
          </div>
        );
    }
}

function mapStateToProps({ auth, job,application }) {
    return {
      auth,
      application,
    
      
    };
  }
  
  export default connect(mapStateToProps)(Job1);


