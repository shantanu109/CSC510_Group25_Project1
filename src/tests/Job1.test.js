import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import Job1 from '../components/Job1';
import { configureStore } from '../store'


const store = configureStore()



const menu = {
ingredients: [
{
inventory_id: {
"itemname": "Lettuce"
},
quantity: 2,
},
]
};




// const application = []


test('render', () => {
const wrapper = mount(
<Provider store={store}>
<Job1 menu = {menu}/>
</Provider>
);
expect(wrapper).toMatchSnapshot();


});
