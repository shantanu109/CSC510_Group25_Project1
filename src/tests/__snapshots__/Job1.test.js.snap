// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Job1)
    menu={
      Object {
        "ingredients": Array [
          Object {
            "inventory_id": Object {
              "itemname": "Lettuce",
            },
            "quantity": 2,
          },
        ],
      }
    }
  >
    <Job1
      application={Array []}
      auth={
        Object {
          "error": null,
          "inProgress": false,
          "isLoggedIn": false,
          "user": Object {},
        }
      }
      dispatch={[Function]}
      menu={
        Object {
          "ingredients": Array [
            Object {
              "inventory_id": Object {
                "itemname": "Lettuce",
              },
              "quantity": 2,
            },
          ],
        }
      }
    >
      <div
        className="post"
        style={
          Object {
            "marginLeft": "50px",
            "width": "50vw",
          }
        }
      >
        <div
          className="post-header"
        >
          <div>
            <h4
              style={
                Object {
                  "display": "inline-block",
                  "marginTop": "-12px",
                }
              }
            >
              Dish : 
            </h4>
            <span
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            />
          </div>
          <div>
            <h4
              style={
                Object {
                  "display": "inline-block",
                  "marginTop": "-12px",
                }
              }
            >
              Price : 
            </h4>
            <span
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            />
          </div>
          <div>
            <h4
              style={
                Object {
                  "display": "inline-block",
                  "marginTop": "-12px",
                }
              }
            >
              Ingredients
            </h4>
            <div
              className="ingredient_container"
            >
              <div
                className="ingredient_tab"
              >
                <div
                  className="field"
                >
                  <span
                    style={
                      Object {
                        "fontWeight": "bold",
                        "marginLeft": "10px",
                      }
                    }
                  >
                    No
                  </span>
                </div>
                <div
                  className="field"
                >
                  <span
                    style={
                      Object {
                        "fontWeight": "bold",
                        "marginLeft": "10px",
                      }
                    }
                  >
                    Name
                  </span>
                </div>
                <div
                  className="field"
                >
                  <span
                    style={
                      Object {
                        "fontWeight": "bold",
                        "marginLeft": "10px",
                      }
                    }
                  >
                    Quantity
                  </span>
                </div>
              </div>
              <div
                className="ingredient_tab"
              >
                <div
                  className="field"
                  key="0"
                >
                  <span
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  >
                    1
                  </span>
                </div>
                <div
                  className="field"
                  key="0"
                >
                  <span
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  >
                    Lettuce
                  </span>
                </div>
                <div
                  className="field"
                  key="0"
                >
                  <span
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  >
                    2
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Job1>
  </Connect(Job1)>
</Provider>
`;
